File: BlueQuartz/5100WG/branches/dev/ui/base-dns.mod/glue/handlers/timezone.pl
#!/usr/bin/perl -w
# $Id: timezone.pl 201 2003-07-18 19:11:07Z will $
# Copyright 2000, 2001 Sun Microsystems, Inc., All rights reserved.
#
# Detects self-reference in DNS servers, enables named accordingly

my($timezone, $bindzone) = ('/etc/localtime', '/var/lib/named/etc/localtime');

my $DEBUG = 0;
$DEBUG && open(STDERR, ">>/tmp/dns_timezone");
$DEBUG && warn `date` .' '. $0;

use lib qw( /usr/sausalito/perl );
use CCE;
$cce = new CCE;
$cce->connectfd();

use File::Copy;

my @sysoids = $cce->find('System');

unlink($bindzone);
copy($timezone, $bindzone);
chmod(0644, $bindzone);

# hup named to pickup the updated timezone
my $time = time();
my $ok = $cce->set($sysoids[0], 'DNS', {'commit' => $time});

$cce->bye('SUCCESS');
exit 0;

# Copyright (c) 2003 Sun Microsystems, Inc. All  Rights Reserved.
# 
# Redistribution and use in source and binary forms, with or without 
# modification, are permitted provided that the following conditions are met:
# 
# -Redistribution of source code must retain the above copyright notice, 
# this list of conditions and the following disclaimer.
# 
# -Redistribution in binary form must reproduce the above copyright notice, 
# this list of conditions and the following disclaimer in the documentation  
# and/or other materials provided with the distribution.
# 
# Neither the name of Sun Microsystems, Inc. or the names of contributors may 
# be used to endorse or promote products derived from this software without 
# specific prior written permission.
# 
# This software is provided "AS IS," without a warranty of any kind. ALL EXPRESS OR IMPLIED CONDITIONS, REPRESENTATIONS AND WARRANTIES, INCLUDING ANY IMPLIED WARRANTY OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE OR NON-INFRINGEMENT, ARE HEREBY EXCLUDED. SUN MICROSYSTEMS, INC. ("SUN") AND ITS LICENSORS SHALL NOT BE LIABLE FOR ANY DAMAGES SUFFERED BY LICENSEE AS A RESULT OF USING, MODIFYING OR DISTRIBUTING THIS SOFTWARE OR ITS DERIVATIVES. IN NO EVENT WILL SUN OR ITS LICENSORS BE LIABLE FOR ANY LOST REVENUE, PROFIT OR DATA, OR FOR DIRECT, INDIRECT, SPECIAL, CONSEQUENTIAL, INCIDENTAL OR PUNITIVE DAMAGES, HOWEVER CAUSED AND REGARDLESS OF THE THEORY OF LIABILITY, ARISING OUT OF THE USE OF OR INABILITY TO USE THIS SOFTWARE, EVEN IF SUN HAS BEEN ADVISED OF THE POSSIBILITY OF SUCH DAMAGES.
# 
# You acknowledge that  this software is not designed or intended for use in the design, construction, operation or maintenance of any nuclear facility.

File: BlueOnyx/5210R/platform/alpine.mod/ci/application/libraries/uifc/TimeZone.php
<?php
// Author: Kenneth C.K. Leung
// $Id: TimeZone.php

global $isTimeZoneDefined;
if($isTimeZoneDefined)
  return;
$isTimeZoneDefined = true;

include_once("uifc/FormField.php");
include_once("uifc/FormFieldBuilder.php");

//
// protected class variables
//

class TimeZone extends FormField {
  //
  // public methods
  //

  // Sets the current label
  function setCurrentLabel($label) {
    $this->Label = $label;
  }

  // Returns the current label
  function getCurrentLabel() {
    if (!isset($this->Label)) {
      $this->Label = "";
    }
    return $this->Label;
  }

  // Sets the current label-description:
  function setDescription($description) {
    if (!isset($this->Description)) {
      $this->Description = "";
    }
    $this->Description = $description;
  }

  // Returns the current label-description:
  function getDescription() {
    return $this->Description;
  }

  function toHtml($style = "") {

    $page =& $this->getPage();
    $form =& $page->getForm();
    $formId = $form->getId();
    $id = $this->getId();
    $i18n =& $page->getI18n();
    $value = $this->getValue();
//print_rp($page);
    if (isset($page->BXLabel[$id])) {
      $LabelArray = array_keys($page->BXLabel[$id]);
      $HtmlLabel = $i18n->getHtml($LabelArray[0]);
      $DescArray = array_values($page->BXLabel[$id]);
      $HtmlDesc = $i18n->getWrapped($DescArray[0]);
    }

    if (!isset($HtmlLabel)) {
      $HtmlLabel = $this->getCurrentLabel();
    }

    if (!isset($HtmlDesc)) {
      $HtmlDesc = $this->getDescription();
    }

    // make sure there is a default value
    $value = ($value == "") ? "US/Eastern" : $value;

    $html_out = '

            <fieldset class="label_side top">
                    <label for="' . $id . '" title="' . $HtmlDesc . '" class="tooltip right uniform">' . $HtmlLabel . '<span></span></label>
                    <div>
                    ' . ln_display_timezone_selector($value) . '
                    </div>
            </fieldset>


            ';

  return $html_out;

  }

}

/*
Copyright (c) 2014 Michael Stauber, SOLARSPEED.NET
Copyright (c) 2014 Team BlueOnyx, BLUEONYX.IT
Copyright (c) 2003 Sun Microsystems, Inc. 
All Rights Reserved.

1. Redistributions of source code must retain the above copyright 
   notice, this list of conditions and the following disclaimer.

2. Redistributions in binary form must reproduce the above copyright 
   notice, this list of conditions and the following disclaimer in 
   the documentation and/or other materials provided with the 
   distribution.

3. Neither the name of the copyright holder nor the names of its 
   contributors may be used to endorse or promote products derived 
   from this software without specific prior written permission.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS 
"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT 
LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS 
FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE 
COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, 
INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT 
LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN 
ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
POSSIBILITY OF SUCH DAMAGE.

You acknowledge that this software is not designed or intended for 
use in the design, construction, operation or maintenance of any 
nuclear facility.

*/
?>